docs/user-guide/mc.ipynb

```diff
@@ -1,885 +0,0 @@
-    "\n",
-    "### Random number generators and seeding\n",
-    "\n",
-    "By default, `mc.run` constructs its own random number generator (RNG) and seeds it from the system RNG.\n",
-    "If you want to make the process reproducible, you can pass a concrete seed via the `seed` argument.\n",
-    "See the documentation of [mc.run](../generated/modules/scippuncertainty.mc.driver.run.rst) for details.\n",
-    "\n",
-    "`mc.run` also sends a log message with the seed it uses.\n",
-    "We can see it by configuring the logger, e.g., using the following."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "4a83e049-2ee5-4bfd-b58e-7b467c09c22a",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "import logging\n",
-    "from scippuncertainty.logging import get_logger\n",
-    "\n",
-    "handler = logging.StreamHandler()\n",
-    "handler.setLevel(\"INFO\")\n",
-    "get_logger().addHandler(handler)\n",
-    "get_logger().setLevel(\"INFO\")"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "aa00c433-dd47-4947-9fbb-173c96ad0462",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "results = mc.run(\n",
-    "    compute_average_speed,\n",
-    "    samplers={\"pos\": pos_sampler, \"time\": time_sampler},\n",
-    "    accumulators={\"speed\": speed_accumulator,\n",
-    "                  \"average_speed\": average_speed_accumulator},\n",
-    "    n_samples=10,\n",
-    "    progress=False,\n",
-    ")"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "id": "603de723-cd46-47df-a188-2974adb1fb4d",
-   "metadata": {},
-   "source": [
-    "### Multi-threading\n",
-    "\n",
-    "`mc.run` can use multiple threads where each thread processes a subset of samples.\n",
-    "Simply set the desired number of threads with the `n_threads` argument.\n",
-    "\n",
-    "But bear in mind that this uses Python's threads.\n",
-    "So you will only see a speedup if your function spends a lot of time in code that releases the global interpreter lock (GIL).\n",
-    "Most functions in Scipp do this.\n",
-    "However, since the data in the examples here is so small, multi threading does not improve performance.\n",
-    "\n",
-    "Note further that most functions in Scipp are internally multi threaded.\n",
-    "So you should make sure not to oversubscribe your CPU.\n",
-    "\n",
-    "### Skipping samples\n",
-    "\n",
-    "Monte-Carlo can sometimes produce samples that our operation cannot handle, for example because a sample contains a 0 or a fit fails to converge\n",
-    "It is possible to skip those samples.\n",
-    "\n",
-    "For example, suppose that we had a function that requires positive inputs like `sqrt`.\n",
-    "We can detect negative inputs and return `mc.SkipSample` instead of actual data.\n",
-    "This instructs `mc.run` to ignore this sample and carry on with the next one."
```

Comment:
Feels a bit odd to handle this here. Wouldn't it be more logical to handle this in the sampler (via a custom optional function), to effectively modify the distribution?

Reply:
In this case, possibly. I initially added this feature because some fits were failing to converge. This cannot be detected by the sampler. So the example is not the best, but it is difficult to construct something more realistic.